File: day1/puzzle2.py
#!/usr/bin/env python3

data_file_name = "puzzle1_data.txt"

with open(data_file_name, 'r') as file:
    depth_measurements = [int(line.rstrip()) for line in file]

sliding_measurements = []

for i, measurement in enumerate(depth_measurements):
    if i < 2: # Guard against index out of bounds
        continue

    minus_2 = depth_measurements[i-2]
    minus_1 = depth_measurements[i-1]

    sliding_measurements.append(measurement + minus_1 + minus_2)

# Same logic as puzzle 1
deeper_count = 0
for i, measurement in enumerate(sliding_measurements):
    if i < 1: # Guard against index out of bounds
        continue

    if measurement > sliding_measurements[i - 1]:
        deeper_count += 1

print(deeper_count)

File: day1/puzzle2.swift
#!/usr/bin/swift

import Foundation

let dataFilePath = "day1_data.txt"

// Read file
let fileContents = try String(contentsOfFile: dataFilePath, encoding: String.Encoding.utf8)
let depthMeasurements = fileContents.split(whereSeparator: \.isNewline).map( { Int($0)! } ) // Force unwrapping not ideal

// First make sure there are 2 values before the current (prevent indexOutOfRange error), then sum current and previous two values
let slidingMeasurements = depthMeasurements.enumerated().filter( { $0.offset > 1 } ).map( { $0.element + depthMeasurements[$0.offset - 1] + depthMeasurements[$0.offset - 2] } )

// Same logic as puzzle 1 - filter by looking at previous measurement (guard against first value, otherwise IndexOutOfRange error)
let deeperMeasurements = slidingMeasurements.enumerated().filter( { ($0.offset != 0) ? $0.element > slidingMeasurements[$0.offset - 1] : false } )
print(deeperMeasurements.count)